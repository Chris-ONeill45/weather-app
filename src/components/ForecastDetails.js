import React from "react";

function ForecastDetails({ forecast }) {
  const { date, temperature, humidity, wind } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <div className="forecast-details__date ">{formattedDate}</div>
      <div className="forecast-details__min-temperature">
        Min Temperature: {temperature.min}&deg;C
      </div>
      <div className="forecast-details__max-temperature">
        Max Temperature: {temperature.max}&deg;C
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}%</div>
      <div className="forecast-details__wind-speed">
        Wind Speed: {wind.speed} mph
      </div>
      <div className="forecast-details__wind-direction">
        Wind Direction: {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;
