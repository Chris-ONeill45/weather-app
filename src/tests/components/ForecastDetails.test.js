import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const mockForecast = {
    date: 1111111,
    humidity: 50,
    temperature: {
      min: 10,
      max: 20,
    },
    wind: {
      speed: 10,
      direction: "NW",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={mockForecast} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders with correct data", () => {
    const { getByText } = render(<ForecastDetails forecast={mockForecast} />);

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-details__date");
    expect(getByText("Min Temperature: 10°C")).toHaveClass(
      "forecast-details__min-temperature"
    );
    expect(getByText("Max Temperature: 20°C")).toHaveClass(
      "forecast-details__max-temperature"
    );
    expect(getByText("Humidity: 50%")).toHaveClass(
      "forecast-details__humidity"
    );
    expect(getByText("Wind Speed: 10 mph")).toHaveClass(
      "forecast-details__wind-speed"
    );
    expect(getByText("Wind Direction: NW")).toHaveClass(
      "forecast-details__wind-direction"
    );
  });
});
