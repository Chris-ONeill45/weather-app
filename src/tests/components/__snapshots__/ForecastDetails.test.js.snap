// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date "
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__min-temperature"
    >
      Min Temperature: 10°C
    </div>
    <div
      class="forecast-details__max-temperature"
    >
      Max Temperature: 20°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 50%
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      Wind Speed: 10 mph
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      Wind Direction: NW
    </div>
  </div>
</DocumentFragment>
`;
